### PR TITLE
[FEATURE] Manage lock & remove command limit management

### PR DESCRIPTION
**Pull request by pdenis**

### CODE REVIEW

File: DomainManager/LockManager.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    /**
+     * @var LockRepository
+     */
+    protected $lockRepository;
+    /**
+     * @param string          $class
+     * @param ObjectManager   $om
+     */
+    public function __construct($class, ObjectManager $om, LockRepository $lockRepository)
```
Comment (ghost) [inline DomainManager/LockManager.php:34]: phpdoc :p

File: Command/ProcessEventCommand.php [hunk 2]
```diff
@@ -53,34 +67,45 @@ protected function configure()
...
-                $this->eventManager->save($event);
+
             }
+        } catch(\Exception $e) {
+            $exception = $e;
+        }
+        $this->lockHandler->release();
+
+        if ($exception) {
+            throw $exception;
```
Comment (ghost) [inline Command/ProcessEventCommand.php]: déporter dans le try et catch, ça me semble plus propre

Comment (juchi) [inline Command/ProcessEventCommand.php]: I think we need to `$this->lockHandler->release();` but we don't have "finally" in php 5.4...

File: Command/ProcessEventCommand.php [hunk 3]
```diff
@@ -53,34 +67,45 @@ protected function configure()
...
+     * @throws LockException
+     * @throws \Exception
+     *
      * @return int|null|void
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $events = $this->eventRepository->findBy(array(), array('createdAt' => 1), $input->getArgument('n'));
+        if ($this->lockHandler->isLocked()) {
+            throw new LockException('Command is locked by another processus');
```
Comment (juchi) [inline Command/ProcessEventCommand.php]: processus => process

File: Repository/EventRepository.php [hunk 4]
```diff
@@ -10,4 +10,13 @@
...
 class EventRepository extends DocumentRepository
 {
+    /**
+     * @return Event
+     */
+    public function findFirstTodoEvent()
+    {
+        $events = $this->findBy(array(), array('createdAt' => 1), 1);
+
+        return array_pop($events);
```
Comment (juchi) [inline Repository/EventRepository.php:20]: can't we use findOneBy directly ?
edit: no we can't

**Comment by juchi** [comment]
:+1:

**Pull request merged**